Automattic/wp-calypso
# Agency Program: Add Survey banner in Pro Dashboard and Partner portal licenses page.

This PR adds the Survey banner for the Jetpack Agency Program in the Dashboard and Licenses screen.

<img width="1376" alt="Screen Shot 2023-06-16 at 6 01 24 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/7400265a-ee95-462f-94cc-7f46b85356b9">


Related to 1202619025189113-as-1204823653826855

## Proposed Changes
* Implement a Site Survey Banner component for the Dashboard and Licenses screen.
* Make sure the Site survey banner only appears on the page with an English locale.

## Testing Instructions

1. Use the Jetpack cloud live link and goto `/dashboard` or `/partner-portal/licenses`
2. Click the **'Take survey'** button and confirm you are redirected to https://automattic.survey.fm/agency-partnership-usage-survey
3. Reset **jetpack-dashboard-agency-program-survey-banner-preference** user preference and repeat step 1. (See below how to reset preferences)
4. Click the **dismiss** button and confirm the banner disappears.

***How to reset preference?***
1. You will need to be on staging site to access the reset functionality. (Go to  https://cloud.jetpack.com and make sure you are running behind proxy)
2. In the Calypso page at the bottom right. Hover on the debug icon and click preferences.
<img width="523" alt="Screen Shot 2023-06-16 at 6 07 42 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/e81bf552-9593-4d23-b177-21315b67b1f4">

3. Look for **jetpack-dashboard-survey-banner-preference** and click **x** button to clear the preference.
<img width="583" alt="Screen Shot 2023-06-16 at 6 09 06 PM" src="https://github.com/Automattic/wp-calypso/assets/56598660/c376781a-a8f4-42b7-9d10-b967af36636f">

***Test other locale***
1. Switch your language to non-English in https://wordpress.com/me/account.
2. Go back to Agency dashboard and confirm that the survey banner does not appear.

## Pre-merge Checklist
